packages/apps/correios/lib-mjs/calculate-shipping.mjs

```diff
@@ -0,0 +1,398 @@
+              break;
+            case 'm':
+              dimensionValue = dimension.value * 100;
+              break;
+            case 'mm':
+              dimensionValue = dimension.value / 10;
+              break;
+            default:
+              dimensionValue = dimension.value;
+          }
+          // add/sum current side to final dimensions object
+          if (dimensionValue) {
+            const pkgDimension = pkg.dimensions[side];
+            for (let i = 0; i < quantity; i++) {
+              if (!pkgDimension.value) {
+                pkgDimension.value = dimensionValue;
+              } else if (nextDimensionToSum === side) {
+                pkgDimension.value += dimensionValue;
+                if (nextDimensionToSum === 'length') {
+                  nextDimensionToSum = 'width';
+                } else {
+                  nextDimensionToSum = nextDimensionToSum === 'width' ? 'height' : 'length';
+                }
+              } else if (pkgDimension.value < dimensionValue) {
+                pkgDimension.value = dimensionValue;
+              }
+            }
+          }
+        }
+      });
+    }
+  });
+
+  let correiosResult;
+  let docId;
+  let correios;
+
+  try {
+    correios = await newCorreios();
+  } catch (err) {
+    err.app = '[calculate Correios]';
+    logger.error(err);
+    const message = err.message || '[calculate Correios] Contract not found';
+    return {
+      error: 'CALCULATE_ERR',
+      message,
+    };
+  }
+
+  try {
+    const docParams = {
+      cepOrigem,
+      cepDestino: parseZipCode(cepDestino),
+      psObjeto: parserWeight(pkg.weight.value),
+      correios,
+      serviceCodes,
+    };
+
+    docId = getDocId(docParams);
+    const docSnapshot = await getFirestore().doc(docId).get();
+    let docData;
+    if (docSnapshot.exists) {
+      docData = docSnapshot.data();
+      const expiresIn = 1000 * 60 * 60 * 24 * 30 * (metafields?.correiosMonthsExpireDb || 3);
+      const now = Timestamp.now().toMillis();
+      if ((docData.createdAt.toMillis() + expiresIn) > now) {
+        correiosResult = docData.data;
+      }
+    }
+  } catch (err) {
+    logger.error(new Error(`[calculate Correios] Cannot get doc ID ${docId}`));
+    logger.error(err);
+  }
+
+  // fallback Calculate
+  if (!correiosResult) {
+    try {
+      const { data } = await calculateV2({
+        correiosParams: {
+          cepOrigem,
+          cepDestino,
+          psObjeto: pkg.weight.value,
+          comprimento: pkg.dimensions.length.value,
+          altura: pkg.dimensions.height.value,
+          largura: pkg.dimensions.width.value,
+          vlDeclarado,
+          servicosAdicionais,
+        },
+        serviceCodes,
+        correios,
+      });
+      correiosResult = data;
+
+      // save in database
+      if (docId) {
+        getFirestore().doc(docId)
+          .set({
+            data,
+            createdAt: Timestamp.fromDate(new Date()),
```

Comment:
Mudou o "schema" do documento salvo, mudou em todos os lugares?

Reply:
Eu dava uma "resumida" nele pra cortar megabytes mesmo https://cloud.google.com/firestore/pricing 🙃

Reply:
Quer dizer, pelo jeito você ignorou o https://github.com/ecomplus/app-correios-db/blob/master/functions/lib/integration/doc-parsers.js , mas meio que ele tinha uma razão de existir também, ideal seria mantê-lo aqui mas adaptado ao formato novo da resposta, claro...

Reply:
Ah sim,  mudei em todos os lugares sim. Pode manter ou voltamos ?

Reply:
Ah o certo era manter relativo a implementação original porque pra mudar depois vai dar canseira, e salvando a resposta repetida "as is" desse jeito aí vamos pagar (muito) mais pelos nomes dos fields que o Correios usa do que propriamente pelos valores (a parte útil)

Reply:
Ah sim, entendi.